extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -76,16 +82,43 @@ class AmpInstagram extends AMP.BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
+  prerenderCallback() {
+    const image = new Image();
```

Comment:
hmm. Yeah this would take care of managing the resource correctly. 

Not sure if for the 2nd issue you mean by styling the `img` element inside `amp-img`? But yeah possibly can be done.